### PR TITLE
Feature: Add Clean Up Handler to Executor

### PR DESCRIPTION
**Pull request by williams-jack**
## Feature/Problem Description
During the process of running a grading job on Orca, if a container is created _but not started_, then it will need to be cleaned up from the local Docker registry. To ensure this happens, a clean up handler is installed into the grading job executor, which the docker version of this class takes advantage of.

## Solution (Changes Made)
* `_cleanup_handler` added on class `GradingJobExecutor`
  * `DockerGradingJobExecutor` implements this method and removes the container if it is found to exist.
*`InvalidWorkerStateException` thrown if container exists and could not be cleaned up, as this means we may not be able to clean up resources successfully on the worker.

### CODE REVIEW

**Comment by blerner** [comment]
Shouldn't the tests be passing?

File: worker/orca_grader/__main__.py [hunk 1]
```diff
@@ -145,6 +145,8 @@ def run_grading_job(grading_job: GradingJobJSON, no_container: bool,
             handle_grading_job(grading_job, image_name, container_command)
         else:
             handle_grading_job(grading_job, image_name)
+    except InvalidWorkerStateException as ive:
```
Comment (blerner) [inline worker/orca_grader/__main__.py:148]: Where do you get the initials "i v e" from Invalid Worker State Exception?

Comment (blerner) [reply worker/orca_grader/__main__.py:148]: Probably ought to LOGGER.critical this error?  Or do you log every time you raise such an error?

**Comment by williams-jack** [comment]
> Shouldn't the tests be passing?

They will once we merge in the other PR

File: worker/orca_grader/__main__.py (hunk 1 above)
Comment (williams-jack) [reply worker/orca_grader/__main__.py:148]: It gets logged at the `process_db_jobs` function, which this bubbles up to

**Pull request merged**